### ISSUE TITLE
Does not seem to work with Python 3.11 and Django 4.2.1

### ISSUE THREAD

**Issue by nessita** [schinckel/django-shared-property#21]
* Django Shared Property version: main
* Python version: 3.11
* Operating System: Ubuntu

### Description

Trying to use `django-shared-property` with Python 3.11 and Django 4.2.1 with a minimal model like the example in the README results in the following exception:

```
======================================================================
ERROR: testapp.tests (unittest.loader._FailedTest.testapp.tests)
----------------------------------------------------------------------
ImportError: Failed to import test module: testapp.tests
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/home/nessita/projects/django-shared-property/py311test/testapp/tests.py", line 3, in <module>
    from .models import Person
  File "/home/nessita/projects/django-shared-property/py311test/testapp/models.py", line 6, in <module>
    class Person(models.Model):
  File "/home/nessita/projects/django-shared-property/py311test/testapp/models.py", line 10, in Person
    @shared_property
     ^^^^^^^^^^^^^^^
  File "/home/nessita/projects/django-shared-property/src/django_shared_property/decorator.py", line 68, in __init__
    self.parsed = Parser(func)
                  ^^^^^^^^^^^^
  File "/home/nessita/projects/django-shared-property/src/django_shared_property/parser.py", line 110, in __init__
    self.code = compile(self.ast, mode="exec", filename=self.file["filename"])
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: AST node line range (10, 1) is not valid
``` 

### What I Did

I created a new Django project with a `testapp` in it. Running the tests below failed with the reported exception.

* `models.py`

```python
from django.db import models
from django.db.models.functions import Concat
from django_shared_property.decorator import shared_property


class Person(models.Model):
    first_name = models.TextField()
    last_name = models.TextField()

    @shared_property
    def full_name(self):
        return Concat(models.F('first_name'), models.Value(' '), models.F('last_name'))
```

* `tests.py`:
```python
from django.test import TestCase

from .models import Person


class PersonTestCase(TestCase):

    def test_one(self):
        p = Person.objects.create(first_name='Jane', last_name='Doe')
        self.assertEqual(p.full_name, 'Jane Doe')
```

**Comment by schinckel** [comment]
Weird, because I’m running these versions in tox.

**Comment by nessita** [comment]
> Weird, because I’m running these versions in tox.

It surprised me too after I saw the `tox` definitions. Can I help debug in any way?

**Comment by schinckel** [comment]
Sorry, I forgot about this issue. Was able to reproduce it in my local tox: I am on a different machine, so I wonder if that was it. (same OS/architecture though, so that's weird).

**Comment by schinckel** [comment]
Figured it out: Python 3.11 complains when there is a missing end_lineno, or if that one is less than the start lineno.

**Comment by schinckel** [comment]
Fixed by #33
